### DISCUSSION TITLE
Fitting trick for DCB and its parameter vs mass fits

### DISCUSSION THREAD

**Discussion by ram1123** [ram1123/2l2Q_limitSettingTool#32] [Comments_HIG_23_001] [answered]
> Fig. 33-34-35: it seems that there are jumps in the a1 and n1 parameters of the CB function. Maybe you can get a good fit fixing them to the values predicted by your smooth curve of this first iteration of fits, and just fitting for all the others. If you get a good fit anyway, maybe you also get a better approximation of the other parameters, which are more substantial (like the mean / sigma)

@jialin-guo1  This is really a nice suggestion. Did you get this?

**Comment by jialin-guo1** [comment]
@ram1123  after apply this iteration fitting approach the fit results are like:
<img width="760" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/add5af24-63a5-4659-a20a-4d50c18cf946">
<img width="761" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/4b4a1c62-8218-469f-bab1-05dfdac78924">
<img width="759" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/47dca294-7951-4413-b880-fa5d080335fb">
<img width="761" alt="image" src="https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/c689be1f-e8af-4e36-83ef-1063bba07936">

**Reply by ram1123** [reply]
It's better.

Since the parameters should be constant so I don't expect any fluctuations. Why is there still some fluctuations in the last two plot?

**Reply by jialin-guo1** [reply]
![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/8f3184fb-09a6-4069-b99f-5fb3b3913f71)

![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/53b90e07-0278-4ee0-b2ff-9f85c14afea4)

sorry, Because I forgot to setup n2 to constant

**Reply by ram1123** [reply]
Add the link to your webpage where you keep all fits.

**Reply by jialin-guo1** [reply]
https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/

**Comment by jialin-guo1** [comment]
I found some points of n2 value in resolved case still negative, which need to be positive. I will fix this issue.

**Reply by jialin-guo1** [reply]
Fixed: https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/

**Comment by jialin-guo1** [comment, marked as answer]
# 2018merged case before and after improving fitting comparison

|Before  Iterative fit | After Iterative fit |
|--- | --- |
|![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/c8f73217-95cc-42fd-993c-37a13c85e0bc) | ![](https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/merged/Fit_a1_merged_tag.png) |
| ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/fcbeaff1-7526-4582-a536-cb413e48945e)| ![](https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/merged/Fit_a2_merged_tag.png) |
| ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/5c31f748-f0c6-4922-a0a6-5322be5d07ee)| ![](https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/merged/Fit_n1_merged_tag.png) |
|![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/638ed82c-78d5-409f-a6cf-f97a51d1001a) | ![](https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/merged/Fit_n2_merged_tag.png) |
| ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/cca3ce0b-334b-488e-bf5a-eb0bc4d6e8ec) | ![](https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/merged/Fit_mean_merged_tag.png)     |
| ![image](https://github.com/ram1123/2l2Q_limitSettingTool/assets/68882142/8277c149-1ea5-4d6f-80a9-4c3adfb330c9)| ![](https://guoj.web.cern.ch/guoj/HZZ2L2Q/UL_legacy/resolution/2018/merged/Fit_sigma_merged_tag.png)     |
